### PR TITLE
Harden the ddlogrecord API

### PR DESCRIPTION
**Pull request by mbudiu-vmw**
@lalithsuresh : does this address #601 ?

### CODE REVIEW

**Comment by ryzhyk** [comment]
Shall we merge this? Did @lalithsuresh confirm that the crash is gone with this fix?

**Comment by mbudiu-vmw** [comment]
I think we can merge

**Comment by mbudiu-vmw** [comment]
He didn't confirm

File: java/ddlogapi/DDlogRecord.java [hunk 1]
```diff
@@ -396,6 +388,14 @@ public DDlogRecord getStructField(int index) {
         return fromSharedHandle(DDlogAPI.ddlog_get_struct_field(this.handle, index));
     }
 
+    /**
```
Comment (ryzhyk) [inline java/ddlogapi/DDlogRecord.java:391]: Shall we also add 

```
public static DDlogRecord some(DDlogRecord x)
````

Comment (mbudiu-vmw) [reply java/ddlogapi/DDlogRecord.java:391]: Lalith asked for it

**Pull request merged**